### ISSUE TITLE
Publish the plugin to the JetBrains Plugin Repository

### ISSUE THREAD

**Issue by snicoll** [spring-io/spring-javaformat#163]
To make it easier for IntelliJ IDEA users to use this plugin, we should publish it to the JetBrains Plugin Repository, see [the instructions](https://www.jetbrains.org/intellij/sdk/docs/basics/getting_started/publishing_plugin.html).

As far as I can see, it requires to do so with an account so we should probably create one for the team or something.

**Comment by snicoll** [comment]
Let's increase the priority of this one please. It looks like the Jetbrains team is testing plugins that are published there and the recent breakage in 2020.1 could have been caught that way.

**Comment by vpavic** [comment]
Any updates on this?

Managing plugin updates manually is really a tedious chore, and that's far from being the only benefit of having the plugin published to Jetbrains plugin repo.

**Comment by philwebb** [comment]
I've not had a chance to look into it in any detail yet. I wonder if there's an API for updating a plugin. I don't like the idea of doing it manually for each release.

**Comment by vpavic** [comment]
> I don't like the idea of doing it manually for each release.

I agree that automation is desirable, but isn't it overall much less of a hassle to publish manually compared to having **everyone** go through the manual download and installation?

**Comment by philwebb** [comment]
Yeah, I agree.

**Comment by vpavic** [comment]
It appears that the plugin was published to Jetbrains plugin repo as IntelliJ offered the plugin update to me today.

However, it's a bit unclear whether this was done by the Spring team or someone from the community?
See [here](https://plugins.jetbrains.com/plugin/15847-spring-javaformat) for more details.

**Comment by philwebb** [comment]
It wasn't done by us. Looks like someone called Tony Cody uploaded it.

**Comment by snicoll** [comment]
I'll reach out to JetBrains.

**Comment by vpavic** [comment]
It seems that the plugin was removed from the Jetbrains plugin repo.

However this situation IMO makes it even more important to get the plugin (properly) published to repo as it seems that everyone who has it installed is under a risk of having it updated if someone else publishes a plugin with matching id (or whatever IntelliJ uses to match update candidates).

This opens a possibility for some quite nasty consequences.

**Comment by wilkinsona** [comment]
> This opens a possibility for some quite nasty consequences.

Agreed. Hopefully JetBrains have put something in place that stops this from happening again.

**Comment by snicoll** [comment]
I asked them to remove it but they can't guarantee that it won't happen again. The person who published the jar had to modify it as the metadata are incomplete (I know that first hand trying to publish it yesterday). 

I agree we need to put some priority on this, even if we have to publish it manually initially.

**Comment by snicoll** [comment]
> People who want to use the IDEA plugin need to build their own.

They don't. This plugin is released on Maven Central so you can download it from there, or from repo.spring.io. It's true that it would be better if you could install it from the JetBrains market place, which is what we'll address in this issue.

**Comment by tonycody** [comment]
> > People who want to use the IDEA plugin need to build their own.
> 
> They don't. This plugin is released on Maven Central so you can download it from there, or from repo.spring.io. It's true that it would be better if you could install it from the JetBrains market place, which is what we'll address in this issue.

Plugin support IDEA, submit to the Jetbrains Plugin market, otherwise build your own installation, too much trouble.

**Comment by vpavic** [comment]
Honestly speaking, I'm uneasy about using the plugin at all until it gets uploaded to the plugin market properly (that is, by the Spring team). Last time it was (unofficially) uploaded just showed how easy it is for someone to take over a manually installed plugin. That incident occurred around a year and a half ago now - if I'm counting right, there were 6 releases since so it doesn't seem that uploading it manually from time to time would take so much time.

**Comment by tonycody** [comment]
> Honestly speaking, I'm uneasy about using the plugin at all until it gets uploaded to the plugin market properly (that is, by the Spring team). Last time it was (unofficially) uploaded just showed how easy it is for someone to take over a manually installed plugin. That incident occurred around a year and a half ago now - if I'm counting right, there were 6 releases since so it doesn't seem that uploading it manually from time to time would take so much time.

![image](https://user-images.githubusercontent.com/6773121/179722230-b2f7abb1-acba-4353-8948-050dd8e17209.png)
 
In China, from https://checkstyle.org/eclipse-cs-update-site/ to download the jar package is very slow. The first build took me at least half an hour to pull the JAR package.

**Comment by snicoll** [comment]
@tonycody I've already mentioned that you don't have to build the plugin yourself. Here is a link to make that more obvious: https://repo1.maven.org/maven2/io/spring/javaformat/spring-javaformat-intellij-idea-plugin/0.0.34/

**Comment by scordio** [comment]
I'd love to see the IntelliJ plugin published to the JetBrains repository.

In certain projects where we use both the Maven and the IntelliJ plugins, the former usually gets updated by Dependabot. This sometimes forces a user who touches the codebase afterwards to perform the IDE plugin update immediately, otherwise the `validate` goal of the Maven plugin (enforced in the build) might fail if the IDE did a reformatting that is no longer compatible with the Maven plugin's expectations. 

Obviously, `spring-javaformat:apply` can help in such situations. However, having the IDE pull the update would be very nice.

**Comment by fischermatte** [comment]
@snicoll I briefly read through the comments here. What is the state of this issue now in 2024? Why is this specific plugin not in the market place? Is it because mainting the constant publishing to jetbrains can't be automated?

**Comment by fischermatte** [comment]
> @fischermatte It's mainly that we haven't had the time to do the task. The project is mostly internal to Spring (although folks are welcome to use it) and it's not a high priority for us to get it on the marketplace.

thx @philwebb for quick response. I really like this formatter due to its simplicity. We used prettier-java with spotless so far, but it comes with quite some verbose config.

**Comment by simasch** [comment]
I've used this project with several customers and most of my private projects.
It's simple to install the plugin from the JAR file, but it would be very convenient if it would be updated like every other IntelliJ plugin.
